unit_test/CMakeLists.txt

```diff
@@ -38,210 +38,276 @@ IF (KOKKOS_ENABLE_CUDA)
   KOKKOSKERNELS_INCLUDE_DIRECTORIES(${CMAKE_CURRENT_BINARY_DIR}/cuda)
   KOKKOSKERNELS_INCLUDE_DIRECTORIES(REQUIRED_DURING_INSTALLATION_TESTING ${CMAKE_CURRENT_SOURCE_DIR}/cuda)
 
-  APPEND_GLOB(CUDA_BLAS_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/cuda/Test_Cuda_Blas*.cpp)
-  KOKKOSKERNELS_ADD_UNIT_TEST_EXECUTABLE_AND_TEST(
-    blas_cuda
-    SOURCES
-      Test_Main.cpp
-      ${CUDA_BLAS_SOURCES}
+  IF (KOKKOSKERNELS_DISABLE_BLAS_TESTS)
+    MESSAGE(STATUS "  disabling Cuda BLAS tests")
+  ELSE()
+    APPEND_GLOB(CUDA_BLAS_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/cuda/Test_Cuda_Blas*.cpp)
+    KOKKOSKERNELS_ADD_UNIT_TEST_EXECUTABLE_AND_TEST(
```

Comment:
It may be a little bit clearer to modify `KOKKOSKERNELS_ADD_UNIT_TEST_EXECUTABLE_AND_TEST` to do something like:

````
KOKKOSKERNELS_ADD_UNIT_TEST_EXECUTABLE_AND_TEST(
     blas_cuda
      SOURCES
        Test_Main.cpp
        ${CUDA_BLAS_SOURCES}
      )
     COMPONENTS blas
  ENDIF()
````

And have an option like:
````
-DKokkosKernels_COMPONENTS=blas;sparse
````

Then we could also fix the ETI so that only that blas, sparse, etc get built.
````
KOKKOSKERNELS_GENERATE_ETI(Blas1_abs abs
  COMPONENTS blas
  HEADER_LIST ETI_HEADERS
  SOURCE_LIST SOURCES
  TYPE_LISTS FLOATS LAYOUTS DEVICES
)
````

Does this seem cleaner to you? If so, I can push a draft PR.

Reply:
That's better at least because it would avoid the repetition that I did for every backend/component.
I just did not know about that `COMPONENTS` keyword but it seems quite useful to define something akin to sub-modules/sub-packages within a CMake build.
If it's not too much work for you to update my PR or make one of your own it would be nice.
I think at the moment using this approach for tests is more important than for the library itself.
Ultimately it's nice to have both, especially if some users are only doing dense linear algebra, they might not need/want the sparse and graph stuffs built and installed.